---
title: "UTI Loci PheWAS Browser"
author: "Wayne Monical"
date: "2025-09-17"
output: 
  html_document
---

```{r library_load, echo=FALSE, warning=FALSE, message=FALSE}
library(tidyverse)
library(plotly)
library(DT)
```

## About

Welcome to the Urinary Tract Infection PheWAS browswer. This site displays the phenome-wide association study between 29 loci that were found to be in significant association with urinary tract infections (UTI). This site was built by Wayne Monical in collaboration with Atlas Khan and the Kiryluk lab at Columbia University. All SNP positions are on GRCh37.


```{r, echo=FALSE}

# read in data
phewas = 
  read_csv('data/all_data.csv', show_col_types = FALSE) |> 
  mutate(
    snp = paste(chrom, pos, reference, alternate, sep = "-"),
    beta = round(beta, 4),
    or = round(or, 4),
    se = round(se, 4),
    round(af, 2)) 

# get strongest phenotype association for each snp
best_phewas = 
  phewas |> 
  group_by(snp)  |> 
  filter(p == min(p, na.rm = TRUE))  |> 
  ungroup() |> 
  arrange(chrom, pos) 
```



## Top Loci

```{r, echo = FALSE}
table_cols = c("snp", 'phenostring', "phenocode", "category", 
               "beta","or", "se","p", 
                "num_cases", "num_controls", "af" )


best_phewas |> 
  select( all_of(table_cols)) |> 
  mutate(# put in links
    snp = paste0(
      #"<a href='", base_url, snp, ".html' target='_blank'>", 
      #snp, # This is the text the user sees
      #"</a>" ))|> 
      "<a href='", "variants/", snp, ".html' target='_blank'>", 
      snp, # This is the text the user sees
      "</a>" ))|> 
  datatable(
    colnames = c('SNP', 'Strongest Association','Phenocode',  'Group', 
                'Beta', ' OR', 'SE', 'P-value',
               'Number of Cases', 'Number of Controls','Allele Frequency'),
    escape = FALSE)
```




